src/sonic-yang-models/yang-models/sonic-sflow.yang

```diff
@@ -0,0 +1,142 @@
+module sonic-sflow{
+
+    namespace "http://github.com/Azure/sonic-sflow";
+    prefix sflow;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+    import sonic-port {
+        prefix port;
+    }
+    /*
+    import sonic-vlan {
+        prefix vlan;
+    }
+    */
+    import sonic-portchannel {
+        prefix lag;
+    }
+    /*
+    import sonic-mgmt-port {
+        prefix mgmt-port;
+    }
+    */
+
+    description "SFLOW yang Module for SONiC OS";
+
+    revision 2021-04-26 {
+        description "First Revision";
+    }
+
+    container sonic-sflow {
+
+        container SFLOW_COLLECTOR {
+            list SFLOW_COLLECTOR_LIST {
+                max-elements 2;
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..64;
+                    }
+                }
+
+                leaf collector_ip {
+                    mandatory true;
+                    type inet:ip-address;
+                }
+
+
+                leaf collector_port {
+                    type inet:port-number;
+                    default 6343;
+                }
+
+            } /* end of list SFLOW_COLLECTOR_LIST */
+        } /* end of container SFLOW_COLLECTOR */
+
+        container SFLOW_SESSION {
+            list SFLOW_SESSION_LIST {
+                key "port";
+
+                leaf port {
+                    type union {
+                        type leafref {
+                            path "/port:sonic-port/port:PORT/port:PORT_LIST/port:name";
+                        }
+                        type string {
+                            pattern "all";
+                        }
+                    }
+                }
+
+                leaf admin_state {
+                    type stypes:admin_status;
+                    default up;
+                }
+
+                leaf sample_rate {
+                    type uint32 {
```

Comment:
The sample rate is valid only for interface "port_name" nad not for interface "all". Can we please add the dependency?

Reply:
Addressed